### DISCUSSION TITLE
/usr/lib/node_modules/puppeteer/.local-chromium not found in sail

### DISCUSSION THREAD

**Discussion by eskiesirius** [spatie/browsershot#694] [Bugs]
I followed this 
```
curl -sL https://deb.nodesource.com/setup_14.x | sudo -E bash -
sudo apt-get install -y nodejs gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgbm1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget libgbm-dev libxshmfence-dev
sudo npm install --location=global --unsafe-perm puppeteer
sudo chmod -R o+rx /usr/lib/node_modules/puppeteer/.local-chromium
```
but the problem is that `/usr/lib/node_modules/puppeteer/.local-chromium` not found

**Comment by aquibj0** [comment]
Did you able to solve this?

**Reply by eskiesirius** [reply]
everything is good.. since i am using sail, instead doing the last part. We should manually run the installer.js
```
./vendor/bin/sail bash
node node_modules/puppeteer/installer.js
```

this will create a chromium path in which you must manually specify it by using
`Browsershot::setChromePath(<your chromium path>)`

**Reply by michaelklopf** [reply]
We solved this with a `puppeteerrc.cjs` in the root directory if I'm right

```
const { join } = require("path");

/**
 * @type {import("puppeteer").Configuration}
 */
module.exports = {
    cacheDirectory: join(__dirname, ".cache", "puppeteer"),
};

```

**Reply by JeffreyHosler** [reply]
> everything is good.. since i am using sail, instead doing the last part. We should manually run the installer.js
> 
> ```
> ./vendor/bin/sail bash
> node node_modules/puppeteer/installer.js
> ```
> 
> this will create a chromium path in which you must manually specify it by using `Browsershot::setChromePath(<your chromium path>)`

Do you mind sharing your whole setup? I am getting ``Error: Failed to launch the browser process! spawn /home/sail/.cache/puppeteer/chrome/linux-1095492 EACCES``

Docker File:
```
RUN apt-get update \
    && apt-get install -y gconf-service libasound2 libappindicator3-1 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgbm1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget libgbm-dev libatk-bridge2.0-0 \
    && npm install --global --unsafe-perm puppeteer
```
then i ran
```
./vendor/bin/sail bash
node node_modules/puppeteer/install.js
```
note: installer.js didn't exist for me but install.js did. It returned a patj

Code

```
        Browsershot::url('https://google.com')
            ->setChromePath("/home/sail/.cache/puppeteer/chrome/linux-1095492")
            ->noSandbox()
            ->savePdf('example.pdf');
```

**Reply by booni3** [reply]
I solved this by installing chromium separately, using `sudo apt-get install chromium-browser` and then specifying the chromium location:

```php
Browsershot::url('https://google.com')
    ->setChromePath('/snap/bin/chromium')
    ->bodyHtml()
```

**Reply by aeq-dev** [reply]
Hi @JeffreyHosler 
I'm facing the same issue, did you find a solution please ? thanks
